libqtile/config.py

```diff
@@ -493,6 +493,16 @@ def paint(self, path: str, mode: str | None = None) -> None:
     def gaps(self) -> Iterable[BarType]:
         return (i for i in [self.top, self.bottom, self.left, self.right] if i)
 
+    def finalize_gaps(self) -> None:
+        def remove(attr: str) -> None:
+            gap = getattr(self, attr, None)
+            if gap is not None:
+                gap.finalize()
+                setattr(self, attr, None)
```

Comment:
This line seems to be the only key difference between current and proposed approach (as we could skip scenarios where the gap is `None`).

Why do we need to set this attribute to `None` to solve the issue?

Reply:
Because of this part of the stack trace:
```
  File "/home/nizar/.local/lib/python3.12/site-packages/libqtile/config.py", line 647, in resize
    self._configure(self.qtile, self.index, x, y, w, h, self.group)
  File "/home/nizar/.local/lib/python3.12/site-packages/libqtile/config.py", line 479, in _configure
    i._configure(qtile, self, reconfigure=reconfigure_gaps)
  File "/home/nizar/.local/lib/python3.12/site-packages/libqtile/bar.py", line 331, in _configure
    self.drawer.clear(self.background)
  File "/home/nizar/.local/lib/python3.12/site-packages/libqtile/backend/base/drawer.py", line 315, in clear
    self.ctx.save()
```

resize() -> _configure() -> drawer.clear(), but the drawer has already been .finalized() from earlier. resize() only does this call if the bar is not none, so by deleting the finalized bar from the screen, we can avoid this call chain.

Let me add that to the commit message.